applications/API/models.py

```
@@ -0,0 +1,656 @@
+            # Generate a unique reference_id
+            while True:
+                candidate = f"{secrets.choice(INSULT_REFERENCE_ID_PREFIX_OPTIONS)}_{encode_base64(int(self.insult_id))}"
+                if not type(self).objects.filter(reference_id=candidate).exists():
+                    self.reference_id = candidate  # pyrefly: ignore
+                    break
+            # Only update the reference_id field
+            self.save(update_fields=["reference_id"])
+            return candidate
+
+    def __str__(self) -> str:
+        return f"{self.reference_id} - ({self.category}) - NSFW: {self.nsfw}"
```

Ruff (0.14.0)

Order of model's inner classes, methods, and fields does not follow the Django Style Guide: Magic method should come before custom method

(DJ012)